src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -269,4 +401,843 @@ public static String getParentModule(Class<?> moduleClass)
+        final List<Integer> result = new ArrayList<>(tokens.length);
+        for (int token : tokens) {
+            boolean found = false;
+
+            for (int subtraction : subtractions) {
+                if (subtraction == token) {
+                    found = true;
+                    break;
+                }
+            }
+
+            if (found) {
+                continue;
+            }
+
+            result.add(token);
+        }
+
+        return result;
+    }
+
+    /**
+     * Gets the field with the given name from the given class.
+     *
+     * @param fieldClass the class to get the field from.
+     * @param propertyName the name of the field.
+     * @return the field we are looking for.
+     */
+    public static Field getField(Class<?> fieldClass, String propertyName) {
+        Field result = null;
+        Class<?> currentClass = fieldClass;
+
+        while (!Object.class.equals(currentClass)) {
+            try {
+                result = currentClass.getDeclaredField(propertyName);
+                result.trySetAccessible();
+                break;
+            }
+            catch (NoSuchFieldException ignored) {
+                currentClass = currentClass.getSuperclass();
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Constructs string with relative link to the provided document.
+     *
+     * @param moduleName the name of the module.
+     * @param document the path of the document.
+     * @return relative link to the document.
+     * @throws MacroExecutionException if link to the document cannot be constructed.
+     */
+    public static String getLinkToDocument(String moduleName, String document)
+            throws MacroExecutionException {
+        final Path templatePath = getTemplatePath(moduleName.replace("Check", ""));
+        if (templatePath == null) {
+            throw new MacroExecutionException(
+                    String.format(Locale.ROOT,
+                            "Could not find template for %s", moduleName));
+        }
+        final Path templatePathParent = templatePath.getParent();
+        if (templatePathParent == null) {
+            throw new MacroExecutionException("Failed to get parent path for " + templatePath);
+        }
+        return templatePathParent
+                .relativize(Paths.get(SRC, "xdocs", document))
+                .toString()
+                .replace(".xml", ".html")
+                .replace('\\', '/');
+    }
+
+    /** Utility class for extracting description from a method's Javadoc. */
+    private static final class DescriptionExtractor {
+
+        /** Variable that tells if we are in a CODE_LITERAL token. */
+        private static boolean isInCodeLiteral;
+        /** Variable that tells if we are in an HTML_ELEMENT token. */
+        private static boolean isInHtmlElement;
+        /** Variable that tells if we are in an HREF attribute. */
+        private static boolean isInHrefAttribute;
```

Comment:
IDEA inspections:
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/20544/workflows/29b03a15-8b8d-485a-bd8a-2bd63d1b2aa4/jobs/373704/artifacts
```
Static field <code>isInHrefAttribute</code> may not be initialized during class initialization #loc
...
```
Initialize it with `false` and checkstyle complains it is initialized to default value. Place it back in method and method becomes too complex. I am stuck with this one.

Reply:
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=13407&id=org.checkstyle%3Acheckstyle
Regarding the Sonarqube code smells - the first 2 ones with the brackets are not applicable because line length would become more than 100

Regarding the 2 methods that are too complex `getDescriptionFromJavadoc` and `getDefaultValue`, I don't think there's anything to gain from splitting them.